packages/website/src/components/editor/useSandboxServices.ts

```diff
@@ -71,6 +72,13 @@ export const useSandboxServices = (
           colorMode === 'dark' ? 'vs-dark' : 'vs-light',
         );
 
+        // registerInlayHintsProvider may not be present for older TS versions
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        sandboxInstance.monaco.languages.registerInlayHintsProvider?.(
+          sandboxInstance.language,
+          createTwoslashInlayProvider(sandboxInstance),
+        );
```

Comment:
Without optional chaining it fails on TS 4.3.5

<img src="https://github.com/typescript-eslint/typescript-eslint/assets/61150013/484d6c2e-079a-4cd1-a65a-1758e2aad308" alt="error on TS 4.3.5">